### PR TITLE
Remove the stackframes with unnecessary method calls

### LINKED ISSUE #27867: Panic inside init() method giving improper stacktrace
**Description:**
$subject

**Steps to reproduce:**

- Consider the below code:

```
class Person {
    public  string name;
    public int age;

    function init(string name, int age) {
        self.name = name;
        self.age = age;
        panic error("Error!!!!");
    }

    function getName() returns string {
        return self.name;
    }

    function getAge() returns int {
        return self.age;
    }
}

public function main() {
    Person person = new("Praveen", 27);
}
```
- Following error will be thrown with stacktrace when we run the above program:
```
error: Error!!!!
        at praveen.balProject.0_1_0.Person:init(main.bal:8)
           praveen.balProject.0_1_0.Person:$init$(main.bal:1)
           praveen.balProject.0_1_0:main(main.bal:21)
```
- Expected stacktrace:
```
error: Error!!!!
        at praveen.balProject.0_1_0.Person:init(main.bal:8)
           praveen.balProject.0_1_0:main(main.bal:21)
```
- No need for the additional stacktrace `praveen.balProject.0_1_0.Person:$init$(main.bal:1)`

**Affected Versions:**
```
Ballerina Swan Lake Preview 8
Language specification v2020-12-17
Update Tool 0.8.10
```

### PR DESCRIPTION
**Pull request by gabilang**
Fixes #27867

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/ErrorValue.java [hunk 1]
```diff
@@ -406,11 +407,15 @@ private boolean isEmptyDetail() {
             // Remove java sources for bal stacktrace if they are not extern functions.
             return Optional.empty();
         }
+        if (methodName == GENERATE_METHOD_INIT) {
```
Comment (riyafa) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/ErrorValue.java:410]: We can combine with the previous if statement

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/error/ErrorTest.java [hunk 2]
```diff
@@ -415,6 +415,22 @@ public void testStackTraceWithErrorCauseLocation() {
...
+    @Test
+    public void testErrorStackTrace() {
+        Exception expectedException = null;
+        try {
+            BRunUtil.invoke(errorTestResult, "testErrorStackTrace");
+        } catch (Exception e) {
+            expectedException = e;
+        }
+
+        Assert.assertNotNull(expectedException);
```
Comment (warunalakshitha) [inline tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/error/ErrorTest.java:427]: Shall we move this assertion to bal code.

Comment (warunalakshitha) [reply tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/error/ErrorTest.java:427]: Shall we add a testw ith interops as well.

    void interop() {
foo()
} 
void foo() {
throw Errorcreator.createerrorVale();
}

Comment (warunalakshitha) [reply tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/error/ErrorTest.java:427]: need to have multiple java method levels

Comment (gabilang) [reply tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/error/ErrorTest.java:423]: Addressed with https://github.com/ballerina-platform/ballerina-lang/pull/32156/commits/2b00b4661100f5ab3d019fc2d767f6d8fcdf6cb1

**Comment by gabilang** [comment]
The patch coverage is still low after added relevant test cases because no test cases has been added for the `fileName != null` check and we can't remove this check since there is a possibility to have the value `null` as per Java Api doc.

**Review by warunalakshitha — APPROVED** [review]

**Pull request merged**